### PR TITLE
Eg 82 overall effectiveness

### LINKED ISSUE #82: Overall Effectiveness
**Is your feature request related to a problem? Please describe.**
Both the enemy and friendly pokemons should show whether it is effective overall or not towards the other.

**Describe the solution you'd like**
Create a simple function that checks the overall effectiveness, and display it on both pokemons.

### PR DESCRIPTION
**Pull request by BlackCubes**
## Changes
1. Added a file with functionalities in `effectiveness.js` which handles the overall and moves effectiveness between a defender and attacker pokemon.
2. Added a `enemyPokeType` state in `App.js` which stores the enemy's type to be used for effectiveness.
3. Implemented `overallEffectiveness` function and rendered it in `EnemyPokemon.js` to display if the enemy pokemon is effective against the friendly pokemon.
4. Implemented `overallEffectiveness` and `moveEffectiveness` with rendering in `SinglePokemon.js` to display if the friendly pokemon is effective against the enemy pokemon.
5. Styled the overall-effectiveness for both the enemy and friendly pokemon.

## Purpose
Users should see if one pokemon has an overall effectiveness (and vice-versa) against the other based on types.

## Approach
Since the move-effectiveness functionality in `EnemyPokemon.js` could be reused, the first step was to create an effectiveness utility file, and store all related effectiveness functionalities in it. This way, both `overallEffectiveness` and `moveEffectiveness` could reuse the same functionalities, and `EnemyPokemon.js` and `SinglePokemon.js` could also use those two functions. The last step was to then style the overall-effectiveness.

This would create an engaging and better UI experience for the user.

Closes #82

### CODE REVIEW

File: pokemon-app/src/components/EnemyPokemon.js [hunk 1]
```diff
@@ -2,101 +2,46 @@ import React, { useState, useEffect } from "react";
...
-      return "Ineffective";
-    }
-
-    return "";
-  });
-};
+import {
+  moveEffectiveness,
+  overallEffectiveness,
+} from "../utils/effectiveness";
```
Comment (gonzalespaulb) [inline pokemon-app/src/components/EnemyPokemon.js:11]: Great way to clean up the logic using utils

File: pokemon-app/src/utils/effectiveness.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
+import { capitalize } from ".";
+import { typeJson } from "../components/typeMatchups";
+
+/**
+ * Gets the pokemon's type(s); checks to see if that type is in typeJson
+ * (and gets it if it exists); and transforms the values of strongAgainst,
+ * weakAgainst, resistantTo, and vulnerableTo from a string to an array.
+ * @param {[{}]} pokeType
+ * @returns {[{}]} An array of objects.
```
Comment (kmausolf) [inline pokemon-app/src/utils/effectiveness.js:9]: To specify an array of objects, use `Object[]`

File: pokemon-app/src/components/EnemyPokemon.js [hunk 3]
```diff
@@ -2,101 +2,46 @@ import React, { useState, useEffect } from "react";
...
       setEffectivenessArray(
         moveEffectiveness(friendlyPokeType, enemyPokeMoveTypes)
       );
+
+      setOverallEffectivenessString(
+        overallEffectiveness(friendlyPokeType, enemyPokeType)
+      );
     }
-  }, [enemyPokeMoveTypes, friendlyPokeType]);
+  }, [enemyPokeMoveTypes, friendlyPokeType, enemyPokeType]);
```
Comment (krferrell) [inline pokemon-app/src/components/EnemyPokemon.js:45]: I really like this useEffect. Makes me see more ways to use them than I currently do. Great work.

File: pokemon-app/src/components/EnemyPokemon.js [hunk 4]
```diff
@@ -2,101 +2,46 @@ import React, { useState, useEffect } from "react";
...
   const checkDoesMovesKeyExistInObject = () =>
     Object.prototype.hasOwnProperty.call(enemyPokemon, "moves");
 
   useEffect(() => {
-    if (enemyPokeMoveTypes && friendlyPokeType) {
+    if (
+      enemyPokeMoveTypes.length &&
+      friendlyPokeType.length &&
+      enemyPokeType.length
+    ) {
```
Comment (gonzalespaulb) [inline pokemon-app/src/components/EnemyPokemon.js:35]: Just to clarify, does this check if there are things in the array?

File: pokemon-app/src/utils/effectiveness.js [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+    // Return it as an object of these four properties with the values as an array
+    // (this is so we could check if the enemy pokemon's move(s) are super
+    // effective, ineffective, or neutral against the friendly pokemon).
+    return {
+      strongAgainst: strongAgainst.split(", "),
+      weakAgainst: weakAgainst.split(", "),
+      resistantTo: resistantTo.split(", "),
+      vulnerableTo: vulnerableTo.split(", "),
+    };
+  });
```
Comment (krferrell) [inline pokemon-app/src/utils/effectiveness.js:31]: This is really cool. Nice thinking on this one.

File: pokemon-app/src/utils/effectiveness.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+      // then increment.
+    } else if (
+      typeMatchup.weakAgainst.includes(capitalize(attackerMoveType.name))
+    ) {
+      effectiveCounter += 2;
+    }
+  });
+
+  return effectiveCounter;
+};
```
Comment (kmausolf) [inline pokemon-app/src/utils/effectiveness.js:58]: Pretty cool!  Good job accounting for double weaknesses and resistances.

**Review by krferrell — APPROVED** [review]
Stellar work. Keep it up!

**Review by gonzalespaulb — APPROVED** [review]

**Comment by kmausolf** [comment]
Good PR write-up this time!  The high level description helps me understand if your code is doing what you intend for it to do.

**Review by Lap343 — APPROVED** [review]
Nice job setting the effectiveness and ineffectiveness as components to reuse them rather than rewriting the code.

File: pokemon-app/src/components/EnemyPokemon.js (hunk 4 above)
Comment (BlackCubes) [reply pokemon-app/src/components/EnemyPokemon.js:35]: Yes. The `.length` would come out as 0 if it were empty (same for strings), and using that in an if-statement would evaluate to false in code language i.e. 0 is false while 1 is true. So, there needs to be something lol

File: pokemon-app/src/components/SinglePokemon.js [hunk 7]
```diff
@@ -16,13 +20,35 @@ const SinglePokemon = ({
...
+      setEffectivenessArray(
+        moveEffectiveness(enemyPokeType, friendlyPokeMoveTypes)
+      );
+
+      setOverallEffectivenessString(
+        overallEffectiveness(enemyPokeType, friendlyPokeType)
+      );
+    }
+  }, [friendlyPokeMoveTypes, friendlyPokeType, enemyPokeType]);
+
```
Comment (Gjhnsn) [inline pokemon-app/src/components/SinglePokemon.js:51]: ooooh this looks very interesting

**Review by Gjhnsn — COMMENTED** [review]
Looking great Elias!

**Pull request merged**